crux-mir-comp/src/Mir/Compositional/Override.hs

```diff
@@ -168,7 +168,8 @@ runSpec cs mh ms = do
         Some btpr <- liftIO $ termToType sym sc (SAW.ecType ec)
         expr <- liftIO $ W4.freshConstant sym nameSymbol btpr
         let ev = CreateVariableEvent loc nameStr btpr expr
-        liftIO $ addAssumptions sym (singleEvent ev)
+        ovrWithBackend $ \bak ->
```

Comment:
There's quite a few spots in `runSpec` when `ovrWithBackend` is called. It may be worth consolidating them all into a single one at the top of the function so that you only need to bind `bak` once.

Reply:
There did end up being quite a few; that's a good idea.